skbio/sequence/_sequence.py

```diff
@@ -491,7 +491,23 @@ def _string(self):
 
     def __init__(self, sequence, metadata=None,
                  positional_metadata=None):
-        if isinstance(sequence, Sequence):
+
+        if type(sequence) == np.ndarray or isinstance(sequence, np.ndarray):
+            if sequence.dtype == np.uint8:
+                self._set_bytes_contiguous(sequence)
+            elif sequence.dtype == '|S1':
+                sequence = sequence.view(np.uint8)
+                # Guarantee the sequence is an array (might be scalar before
+                # this).
+                if sequence.shape == ():
```

Comment:
Excellent question. I looked in to it, and interestingly enough `a = np.array([65], dtype=np.uint8)
` has shape `(1, )` but `a = np.array('A', dtype='c')` has shape `()`, so it's not necessary to check in the `np.uint8` case.